docs/machine-learning/how-to-guides/machine-learning-model-predictions-ml-net.md

```diff
@@ -0,0 +1,140 @@
+
+Because the algorithm used in this sample is a linear regression algorithm, the default name of the output column is `Score` which is defined by the [`ColumnName`](xref:Microsoft.ML.Data.ColumnNameAttribute) attribute on the `PredictedPrice` property.
+
+```csharp
+class HousingPrediction : HousingData
+{
+    [ColumnName("Score")]
+    public float PredictedPrice { get; set; }
+}
+```
+
+The `HousingPrediction` data model inherits from `HousingData` to make it easy to visualize the original input data along with the output generated by the model.  
+
+## Set up Prediction Pipeline
+
+Whether making a single or batch prediction, the prediction pipeline needs to be set up. After loading the data preparation pipeline and the trained model, they're  combined into an [`ITransformerChain`](xref:Microsoft.ML.Data.TransformerChain%601).
+
+```csharp
+//Create MLContext 
+MLContext mlContext = new MLContext();
+
+// Load Data Preparation Pipeline and Trained Model
+DataViewSchema dataPrepPipelineSchema, modelSchema;
+ITransformer dataPreparationPipeline = mlContext.Model.Load("data_preparation_pipeline.zip", out modelSchema);
+ITransformer trainedModel = mlContext.Model.Load("model.zip", out modelSchema);
+
+// Combine data preparation pipeline and trained model transformer into TransformerChain
+TransformerChain<ITransformer> predictionPipeline = dataPreparationPipeline.Append(trainedModel);
+```
+
+## Single Prediction
+
+To make a single prediction, create a [`PredictionEngine`](xref:Microsoft.ML.PredictionEngine%602) using the `predictionPipeline` [`ITransformerChain`](xref:Microsoft.ML.Data.TransformerChain%601).
+
+```csharp
+// Create PredictionEngines
+PredictionEngine<HousingData, HousingPrediction> predictionEngine = mlContext.Model.CreatePredictionEngine<HousingData, HousingPrediction>(predictionPipeline);
+```
+
+Then, use the [`Predict`](xref:Microsoft.ML.PredictionEngineBase%602.Predict*) method and pass in your input data as a parameter. Notice that using the [`Predict`](xref:Microsoft.ML.PredictionEngineBase%602.Predict*) method does not require the input to be an [`IDataView`](xref:Microsoft.ML.IDataView)). This is because it conveniently internalizes the input data type manipulation so you can pass in an object of the input data type. Additionally, since `CurrentPrice` is the target or label, don't set a value for it in the input data.
+
+```csharp
+// Input Data
+HousingData inputData = new HousingData
+{
+    Size = 900f,
+    HistoricalPrices = new float[] { 155000f, 190000f, 220000f }
+};
+
+// Get Prediction
+HousingPrediction prediction = predictionEngine.Predict(inputData);
+```
+
+If you access the `Score` property of the `prediction` object, you should get a value similar to `150079`.
+
+## Batch Prediction
+
+Given the following data, load it into an [`IDataView`](xref:Microsoft.ML.IDataView). Because `CurrentPrice` is the target or label, don't set a value for it in the input data.
```

Comment:
>don't set a value [](start = 133, length = 17)

would it be better to say: "no need to set" or "we don't have a value"

Reply:
#resolved